web/src/components/Incidents/processAlerts.ts

```diff
@@ -265,16 +266,16 @@ export function convertToAlerts(
         alertname: alert.metric.alertname,
         namespace: alert.metric.namespace,
         severity: alert.metric.severity as Severity,
-        component: alert.metric.component,
-        layer: alert.metric.layer,
+        component: matchingIncident.component,
+        layer: matchingIncident.layer,
         name: alert.metric.name,
         alertstate: resolved ? 'resolved' : 'firing',
         values: paddedValues,
         alertsStartFiring: firstTimestamp,
         alertsEndFiring: lastTimestamp,
         resolved,
         x: 0, // Will be set after sorting
-        silenced,
+        silenced: matchingIncident.silenced,
```

Comment:
ok to assume the incident is not null but still needs ?? false, in case the old metric doesn't contain the silenced label. edge case, but better to be safe